legacy/src/app/directives/boot_images.js

```diff
@@ -216,17 +221,51 @@ export function maasBootImages(
 
     // Select the default images that should be selected. Current
     // defaults are '18.04 LTS' and 'amd64'.
-    $scope.selectDefaults = function () {
-      angular.forEach($scope.source.releases, function (release) {
-        if (release.name === "bionic") {
-          $scope.source.selections.releases.push(release);
-        }
-      });
-      angular.forEach($scope.source.arches, function (arch) {
-        if (arch.name === "amd64") {
-          $scope.source.selections.arches.push(arch);
+    $scope.selectDefaults = () => {
+      const source = $scope.source;
+      if (
+        source.releases.some((release) => release.name === "bionic") &&
+        source.arches.some((arch) => arch.name === "amd64")
+      ) {
+        $scope.source.selections.osystems = [
+          { osystem: "ubuntu", release: "bionic", arches: ["amd64"] },
+        ];
+      }
+    };
+
+    // Convert resources data returned from BootResource manager into osystem
+    // objects that can be used in the save_ubuntu websocket handler.
+    $scope.convertResourcesToOSystems = (resources) => {
+      if (!resources) {
+        return [];
+      }
+      return resources.reduce((osystems, resource) => {
+        const split = resource.name.split("/");
+        const osName = split[0];
+        const osRelease = split[1];
+        if (resource.rtype === 0) {
```

Comment:
Would be nice to have a resource type enum to make this more meaninful.

Reply:
Done!